### PR TITLE
fix kong mesh navigation

### PR DESCRIPTION
**Pull request by Automaat**
### Description
Fix to showing correct Kong Mesh versions in docs

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions.

### CODE REVIEW

**Review by fabianrbz — APPROVED** [review]

File: app/_data/docs_nav_mesh_dev.yml
Comment (fabianrbz) [inline app/_data/docs_nav_mesh_dev.yml:1]: this file should be renamed to `app/_data/docs_nav_mesh_2.8.x.yml` (assuming 2.8.x is the next version)
also, `release: 2.7.x` should be updated

Comment (Automaat) [reply app/_data/docs_nav_mesh_dev.yml:1]: changed

File: app/_data/kong_versions.yml [hunk 1]
```diff
@@ -389,7 +389,11 @@
   endOfLifeDate: "2025-02-01"
   branch: release-2.6
 - edition: mesh
-  version: preview
+  version: 2.7.0
   release: 2.7.x
+  branch: release-2.7
```
Comment (fabianrbz) [inline app/_data/kong_versions.yml:394]: Is 2.7.x the new latest? If so, we should set latest: true to it, and remove that bit from 2.6.x

Comment (Automaat) [reply app/_data/kong_versions.yml:394]: 2.7.x is not yet released, I will update it in a separate PR, after we release this version

**Review by fabianrbz — APPROVED** [review]

**Comment by Automaat** [comment]
needed for: https://github.com/Kong/docs.konghq.com/pull/7243

**Pull request merged**